content/guides/pedestal-with-component.adoc

```diff
@@ -1,16 +1,419 @@
+----
+
+We'll also need a test system constructor because we don't want to start an
+http server and bind to a port.
+----
+include::component/src/system_test.clj[tags=test-system]
+----
+<1> Create the Pedestal Provider (Jetty Servlet in this case) but
+don't start the server.
+<2> Since we didn't start the server, there's nothing to stop.
+
+Now that we've got our helpers implemented, let's move on to our
+test. Create a test named `greeting-test`.
+----
+include::component/src/system_test.clj[tags=test]
+----
+<1> `sut` will be bound to the started system by `with-system`.
+<2> Use the `service-fn` helper to extract the Pedestal service
+function from the started system.
+<3> Use Pedestal's `response-for` test helper to make a test request
+to the `:greet` route. Use the `url-for` helper to refer to the route
+by name.
+<4> We should get back a '200' status.
+<5> We should get back a response body of 'Hello, world!'
+
+Now let's restart the repl and run our tests.
+----
+boot.user=> (require 'system-test)
+nil
+boot.user=> (clojure.test/run-tests 'system-test)
+
+Testing system-test
+[nREPL-worker-4] INFO io.pedestal.http - {:msg "GET /greet", :line 78}
+
+Ran 1 tests containing 2 assertions.
+0 failures, 0 errors.
+{:test 1, :pass 2, :fail 0, :error 0, :type :summary}
+----
+
+That's it! You now know the fundamentals necessary for implementing
+and testing your Component-based Pedestal services.
+
+== The Whole Shebang
+For reference, here are the complete contents of all the files.
+
+[[app-listing]]
+[source,clojure,subs="-callouts"]
+.src/pedestal.clj
+----
+include::component/src/pedestal.clj[tags=ns]
+
+include::component/src/pedestal.clj[tags=component-init]
+include::component/src/pedestal.clj[tags=component-start]
+include::component/src/pedestal.clj[tags=component-stop]
+
+include::component/src/pedestal.clj[tags=constructor]
+----
+
+
+[[app-listing]]
+[source,clojure,subs="-callouts"]
+.src/routes.clj
+----
+include::component/src/routes.clj[tags=ns]
+
+include::component/src/routes.clj[tags=response]
+
+include::component/src/routes.clj[tags=routes]
+----
+
+
+[[app-listing]]
+[source,clojure,subs="-callouts"]
+.src/system.clj
+----
+include::component/src/system.clj[tags=ns]
+
+include::component/src/system.clj[tags=app]
+
+include::component/src/system.clj[tags=init]
+----
+
+
+[[app-listing]]
+[source,clojure,subs="-callouts"]
+.src/system_test.clj
+----
+include::component/src/system_test.clj[tags=ns]
+
+include::component/src/system_test.clj[tags=url-for]
+
+include::component/src/system_test.clj[tags=service-fn]
+
+include::component/src/system_test.clj[tags=with-system]
+
+include::component/src/system_test.clj[tags=test-system]
+
+include::component/src/system_test.clj[tags=test]
+----
+
+[[app-listing]]
+[source,clojure,subs="-callouts"]
+.build.boot
+----
+include::component/build.boot[]
+----
+== What's next?
+
+In a separate guide (coming soon!), we'll demonstrate how to introduce dependencies
```

Comment:
It's OK to forward-link to a guide you haven't written yet. Just make a placeholder doc with the standard sections.